src/_common/events/event-bus.service.ts

```diff
@@ -0,0 +1,49 @@
+import Vue from 'vue';
+
+export type EventBusDeregister = () => void;
+
+export class EventBus {
+	private e!: Vue;
+
+	private static global = new EventBus();
+
+	constructor() {
+		this.e = new Vue();
+	}
+
+	emit(event: string, ...args: any[]) {
+		this.e.$emit(event, ...args);
+	}
+
+	on(event: string, callback: Function): EventBusDeregister {
+		this.e.$on(event, callback);
+
+		return () => {
+			this.e.$off(event, callback);
+		};
+	}
+
+	once(event: string, callback: Function) {
+		this.e.$once(event, callback);
+	}
+
+	off(event?: string, callback?: Function) {
+		this.e.$off(event, callback);
+	}
+
+	static emit(event: string, ...args: any[]) {
+		this.global.emit(event, ...args);
+	}
+
+	static on(event: string, callback: Function): EventBusDeregister {
+		return this.global.on(event, callback);
+	}
+
+	static once(event: string, callback: Function) {
+		this.global.once(event, callback);
+	}
+
+	static off(event?: string, callback?: Function) {
```

Comment:
let's get rid of the functionality for getting rid of all global events, i think we want this to be as "global" as possible right now. unless we actually do use that in Client maybe?

Reply:
We don't use it. i'm removing.